### PR TITLE
feat(hd-wallet): add package

### PR DESCRIPTION
**Pull request by javadkh2**
This is the first PR related to hd-wallet
check [readme](https://github.com/kadena-community/kadena.js/blob/feat/hd-wallet/packages/libs/hd-wallet/README.md) and [ADRs](https://github.com/kadena-community/kadena.js/tree/feat/hd-wallet/packages/libs/hd-wallet/docs/decisions) for more information

### CODE REVIEW

File: packages/libs/hd-wallet/docs/decisions/0001-use-bip44.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Use BIP44
```
Comment (EnoF) [inline packages/libs/hd-wallet/docs/decisions/0001-use-bip44.md:1]: Maybe somewhere in this decision record, you could also mention something about not supporting `extended-public-keys`

**Review by EnoF — APPROVED** [review]

File: packages/libs/hd-wallet/src/chainweaver/vendor/kadena-crypto.js [hunk 2]
```diff
@@ -8360,8 +8362,8 @@ if (debugUtil && debugUtil.debuglog) {
 }
 /*</replacement>*/
 
-var BufferList = require('./internal/streams/BufferList.js');
-var destroyImpl = require('./internal/streams/destroy.js');
+var BufferList = require('./internal/streams/BufferList');
```
Comment (EnoF) [inline packages/libs/hd-wallet/src/chainweaver/vendor/kadena-crypto.js:8365]: not sure if it matters, but potentially the removal of `.js` makes this extraction `esm` incompatible?

**Review by EnoF — APPROVED** [review]

File: packages/libs/hd-wallet/src/chainweaver/vendor/kadena-crypto.js (hunk 2 above)
Comment (javadkh2) [reply packages/libs/hd-wallet/src/chainweaver/vendor/kadena-crypto.js:8365]: this bundle is created with browserify, I have copied it from the old lib.
it only work if I remove `.js` actually I didnt add them manually my VSCode did it when I move the file to the different path. 
I will check `esm` compatibility

File: .gitignore [hunk 3]
```diff
@@ -96,3 +96,10 @@ temp
 /packages/*/*/test-results
 /packages/*/*/playwright-report
 
+packages/tools/lint-package/src/helpers.js
+packages/tools/lint-package/src/index.js
+packages/tools/lint-package/src/logger.js
+packages/tools/lint-package/src/rules/api-extractor/types.js
+packages/tools/lint-package/src/rules/eslint/config.js
+packages/tools/lint-package/src/rules/package-json/scripts.js
+packages/tools/lint-package/src/rules/typescript/extends.js
```
Comment (alber70g) [inline .gitignore:105]: Would it be better to make a separate PR for this?

Is this happening when running `tsc` in that package? It should, based on tsconfig, create a `lib` for files like this.

Do we maybe need a more generic one? Something like

```
packages/*/*/src/**/*.js
```

File: packages/libs/hd-wallet/.gitignore
Comment (alber70g) [inline packages/libs/hd-wallet/.gitignore:1]: Do we need this file?

File: packages/libs/hd-wallet/package.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      "chainweaver": [
+        "./lib/chainweaver/index.d.ts"
+      ]
+    }
+  },
+  "files": [
+    "lib"
+  ],
+  "scripts": {
+    "build": "tsc && cpy ./src/chainweaver/vendor/**/* ./lib/chainweaver/vendor/",
```
Comment (alber70g) [inline packages/libs/hd-wallet/package.json:29]: Can we add api-extractor to this package?

File: packages/libs/hd-wallet/src/index.ts
Comment (alber70g) [inline packages/libs/hd-wallet/src/index.ts:1]: Is this generic for the other non-legacy/chainweaver implementation as well?

File: packages/libs/hd-wallet/src/chainweaver/vendor/kadena-crypto.d.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+export declare const kadenaMnemonicToRootKeypair: (
+  password: string,
+  mnemonic: string,
+) => Uint8Array;
+
+export declare const kadenaSign: (
+  password: string,
+  message: string,
+  privateKey: string,
+) => Uint8Array;
```
Comment (alber70g) [inline packages/libs/hd-wallet/src/chainweaver/vendor/kadena-crypto.d.ts:28]: Can we change this to 
```suggestion
export declare const kadenaSign: (
  password: string,
  message: string,
  rootKey: string | Uint8Array,
  index: number
) => Uint8Array;
```

File: packages/libs/hd-wallet/src/chainweaver/vendor/kadena-crypto.d.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+export declare const kadenaGenKeypair: (
+  password: string,
+  rootKey: string | Uint8Array,
+  index: number,
+) => [Uint8Array, Uint8Array];
+
+export declare const kadenaGenMnemonic: () => string;
+
+export declare const kadenaGetPublic: (secretKey: Uint8Array) => Uint8Array;
```
Comment (alber70g) [inline packages/libs/hd-wallet/src/chainweaver/vendor/kadena-crypto.d.ts:17]: This function should probably accept the `rootKey`, `password`, `index`

File: .gitignore (hunk 3 above)
Comment (javadkh2) [reply .gitignore:105]: I don't know how they came up here
